lib/puppet/network/http/request.rb

```diff
@@ -19,39 +19,48 @@ def route_into(prefix)
     self.class.new(headers, params, method, path, routing_path.sub(prefix, ''), client_cert, body)
   end
 
-  def format
+  def formatter
     if header = headers['content-type']
       header.gsub!(/\s*;.*$/,'') # strip any charset
       format = Puppet::Network::FormatHandler.mime(header)
-      if format.nil?
-        #TRANSLATORS "mime-type" is a keyword and should not be translated
-        raise _("Client sent a mime-type (%{header}) that doesn't correspond to a format we support") % { header: header }
-      else
-        assert_supported_format(format)
-        return format.name.to_s if format.suitable?
-      end
+
+      return format if valid_network_format?(format)
+
+      #TRANSLATORS "mime-type" is a keyword and should not be translated
+      raise Puppet::Network::HTTP::Error::HTTPUnsupportedMediaTypeError.new(
+              _("Client sent a mime-type (%{header}) that doesn't correspond to a format we support") % { header: headers['content-type'] },
+              Puppet::Network::HTTP::Issues::UNSUPPORTED_MEDIA_TYPE)
     end
 
-    raise _("No Content-Type header was received, it isn't possible to unserialize the request")
+    raise Puppet::Network::HTTP::Error::HTTPBadRequestError.new(
+            _("No Content-Type header was received, it isn't possible to unserialize the request"),
+            Puppet::Network::HTTP::Issues::MISSING_HEADER_FIELD)
   end
 
-  def response_formatter_for(supported_formats, accepted_formats = headers['accept'])
+  def response_formatter_for(supported_formats, default_accepted_formats = nil)
+    accepted_formats = headers['accept'] || default_accepted_formats
+
+    if accepted_formats.nil?
+      raise Puppet::Network::HTTP::Error::HTTPBadRequestError.new(_("Missing required Accept header"), Puppet::Network::HTTP::Issues::MISSING_HEADER_FIELD)
+    end
+
     formatter = Puppet::Network::FormatHandler.most_suitable_format_for(
```

Comment:
Now with formatter method rename similar request - can we change the name of this local variable to avoid confusion with `Puppet::Network::HTTP::Request#formatter` up above?